var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -151,6 +151,7 @@ class Dealii(CMakePackage, CudaPackage):
     depends_on('slepc@:3.6.3',     when='@:8.4.1+slepc+petsc+mpi')
     depends_on('slepc~arpack',     when='+slepc+petsc+mpi+int64')
     depends_on('sundials@:3~pthread', when='@9.0:+sundials')
+    depends_on('trilinos gotype=int', when='+trilinos')
```

Comment:
i don't think Spack can combine `depends_on`, you should probably add this to each `depends_on` below

Reply:
Spack merges all depends_on constraints, so this should be fine